app/Http/Controllers/Front/CheckoutController.php

```diff
@@ -228,9 +228,12 @@ public function postCheckout(Request $request)
                     $attributes = $this->getAttributes($content);
                 }
             } else {
+                
                 $items = new \Illuminate\Support\Collection();
                 $invoiceid = $request->input('invoice_id');
                 $invoice = $this->invoice->find($invoiceid);
+                $processingFee = $payment_method =="razorpay"? 0 : \DB::table(strtolower($request->input('payment_gateway')))->where('currencies',$invoice->currency)->value('processing_fee');
```

Comment:
not a good practice to directly use user input in table name. If this query was something like 
```
\DB::table($request-user_input))->get();
```
A lot of things could have been exploited

Reply:
For now I have handled exception for SQL error. In future we can change Plugins DB Structure to handle such situation.